### DISCUSSION TITLE
Check Permission with React.

### DISCUSSION THREAD

**Discussion by thetheago** [opcodesio/log-viewer#226] [Q&A]
I'm trying to make a validation with LogViewerAuthMiddleware, where I check if my auth()->user() is logged through the barrer token, but when I type localhost/log-viewer I dont have this token.
I found a way to validate, making a button in my aplication (in react) that do a fetch to my /log-viewer passing the token, but the page is returned as a html+js response and I cannot render it, someone have an ideia that how could I do this?

**Comment by arukompas** [comment]
Hey @thetheago , sorry for such a late response! Summer vacations 😄 

In order to provide a Bearer token for authentication, you'll need to publish the Log Viewer layout and modify the headers.
**[You can read about it here](https://log-viewer.opcodes.io/docs/2.x/configuration/access-to-log-viewer#configuring-how-the-front-end-makes-api-requests)**

Because it's a Blade template, you can simply enter the bearer token like so:

```blade
<script>
    window.LogViewer = @json($logViewerScriptVariables);
    // Add additional headers for LogViewer requests like so:
    window.LogViewer.headers['Authorization'] = 'Bearer {{ $user->bearerToken() }}';
</script>
```

Hope this helps!
